src/backend/app/tools.py

```diff
@@ -61,19 +62,79 @@ def generate_prediction(eval_params):
         {h3.geo_to_h3(lat, lon, hex_resolution) for lat, lon in coordinates}
```

Comment:
It seems that `generate_prediction` function is not used anywhere else and can be removed.